scene/gui/graph_node.cpp

```diff
@@ -392,7 +392,7 @@ void GraphNode::_notification(int p_what) {
 				}
 
 				// Draw slot stylebox.
-				if (s.draw_stylebox) {
+				if (s.draw_stylebox && get_child_count()) {
```

Comment:
This would only solve it for a case with no children, but there could still be a mismatch between the number of children and the value of `E.key` I suppose?
So it would at least need to check `E.key < get_child_count()`.

But this is a sign of a deeper issue IMO.

Reply:
@akien-mga , thanks for your comment. During my test, the crash only occurs when deleting the last child of a GraphNode, as well as deleting all child nodes at once. Because delete the last node, the data in the redraw call may not update yet.

and i think the check `E.key < get_child_count()` is done at the begining of the for loop. since the `cache_y.size()` is equal to the child node count.
```
if (E.key < 0 || E.key >= cache_y.size()) {
    continue;
}
```

Reply:
> and i think the check `E.key < get_child_count()` is done at the begining of the for loop. since the `cache_y.size()` is equal to the child node count.

Clearly they are not equal in that moment, otherwise your own check wouldn't achieve much. They aren't guaranteed to be equal anyway, because the cache is only created for visible and valid _Control_-based children, and also not the ones which are set to be top level.

I agree with Akien, that there does seem to be some underlying issue, but I think an ad-hoc fix could also be done for now. It should, however, be a different fix than yours. It should check the validity of child nodes before doing anything with them. In fact, it's just a good idea in general.

See the code where the lines around https://github.com/godotengine/godot/blob/5d986cb8cccc3c6bd980d27b4413e6109534ee38/scene/gui/graph_node.cpp#L252 for a reference.

Reply:
Hi, @YuriSizov , thansk for your review, in my test, the crash occurs at the code 
[`Control *c = Object::cast_to<Control>(get_child(E.key));`](https://github.com/godotengine/godot/blob/166066d9f74e574b68d9f9d0ba05bf42b8de2131/scene/gui/graph_node.cpp#L396)
in the `get_child` function, so i can't do any further check, any suggestion will be appreciate.

Reply:
That's curious. I'd need to check myself, because `get_child()` should not itself crash for non-existent children, and it already checks for the number of children internally.

Reply:
@YuriSizov, thanks for your professional explanation, and i have updated the commit according to your advice, please take a view, thanks!